app/controllers/nonsipp/memberdetails/SchemeMembersListController.scala

```diff
@@ -61,34 +67,88 @@ class SchemeMembersListController @Inject()(
-      val completedMembers = request.userAnswers.get(MembersDetailsCompletedPages(srn)).getOrElse(Map.empty)
-      val membersDetails = request.userAnswers.membersDetails(srn)
-      val completedMembersDetails = completedMembers.keySet
-        .intersect(membersDetails.keySet)
-        .map(k => k -> membersDetails(k))
-        .toMap
-      if (membersDetails.isEmpty) {
-        logger.error(s"no members found")
-        Redirect(routes.PensionSchemeMembersController.onPageLoad(srn))
-      } else {
+      onPageLoadCommon(srn, page, manualOrUpload, mode)
+    }
+
+  def onPreviousViewOnly(srn: Srn, page: Int, year: String, current: Int, previous: Int): Action[AnyContent] =
+    identifyAndRequireData(srn).async {
+      Future.successful(
+        Redirect(
+          controllers.nonsipp.memberdetails.routes.SchemeMembersListController
+            .onPageLoadViewOnly(srn, page, year, (current - 1).max(0), (previous - 1).max(0))
+        )
+      )
+    }
+
+  def onPageLoadCommon(srn: Srn, page: Int, manualOrUpload: ManualOrUpload, mode: Mode)(
+    implicit request: DataRequest[_]
+  ): Result = {
+    val completedMembers = request.userAnswers.get(MembersDetailsCompletedPages(srn)).getOrElse(Map.empty)
+    val membersDetails = request.userAnswers.membersDetails(srn)
+    val completedMembersDetails = completedMembers.keySet
+      .intersect(membersDetails.keySet)
+      .map(k => k -> membersDetails(k))
+      .toMap
+    if (membersDetails.isEmpty) {
+      logger.error(s"no members found")
+      Redirect(routes.PensionSchemeMembersController.onPageLoad(srn))
+    } else {
 
-        completedMembersDetails.view
-          .mapValues(_.fullName)
-          .toList
-          .zipWithRefinedIndex[Max300.Refined]
-          .map { filteredMembers =>
-            Ok(
-              view(
-                form(manualOrUpload, filteredMembers.size >= maxSchemeMembers),
-                viewModel(srn, page, manualOrUpload, mode, filteredMembers)
+      completedMembersDetails.view
+        .mapValues(_.fullName)
+        .toList
+        .zipWithRefinedIndex[Max300.Refined]
+        .map { filteredMembers =>
+          Ok(
+            view(
+              form(manualOrUpload, filteredMembers.size >= maxSchemeMembers),
+              viewModel(
+                srn,
+                page,
+                manualOrUpload,
+                mode,
+                filteredMembers,
+                viewOnlyUpdated = (mode, request.previousUserAnswers) match {
+                  case (ViewOnlyMode, Some(previousUserAnswers)) =>
+                    val updated = getCompletedOrUpdatedTaskListStatus(
+                      request.userAnswers,
+                      previousUserAnswers,
+                      pages.nonsipp.memberdetails.Paths.personalDetails
+                    ) == Updated
+                    logger.info(s"""[ViewOnlyMode] Status for member details list is ${if (updated) "updated"
```

Comment:
Do you want to keep these logs?

Reply:
Yeah just incase we have issues in the future QA's can supply FE logs to narrow down the issue